torchqdynamics/utils/utils.py

```diff
@@ -22,95 +22,106 @@
 ]
 
 
-def is_ket(x: Tensor) -> Tensor:
-    """Check if a tensor is in state vector format."""
+def is_ket(x: Tensor) -> bool:
+    """Check if a tensor is in state vector format.
+
+    Args:
+        x ( Tensor (..., d, n) ): State vector, linear map or density matrix.
+
+    Returns:
+        `True` if the last dimension `n` is `1`, `False` otherwise.
+    """
     return x.size(-1) == 1
 
 
 def ket_to_bra(x: Tensor) -> Tensor:
-    """Linear map (bra) representation of a state vector (ket).
+    r"""Linear map $\bra\psi$ representation of a state vector $\ket\psi$.
 
     Args:
-        x: Tensor of dimension `(..., d, 1)`.
+        x ( Tensor (..., d, 1) ): State vector.
 
     Returns:
-        Tensor of dimension `(..., 1, d)`.
+        ( Tensor (..., 1, d) ): Corresponding linear map.
     """
     return x.adjoint()
 
 
 def ket_to_dm(x: Tensor) -> Tensor:
-    """Density matrix formed by the outer product of a state vector (ket).
+    r"""Density matrix $\ket\psi\bra\psi$ formed by the outer product of a state vector
+    $\ket\psi$.
 
     Args:
-        x: Tensor of dimension `(..., d, 1)`.
+        x ( Tensor (..., d, 1) ): State vector.
 
     Returns:
-        Tensor of dimension `(..., d, d)`.
+        ( Tensor (..., d, d) ): Corresponding density matrix.
     """
     return x @ ket_to_bra(x)
 
 
-def ket_overlap(x: Tensor, y: Tensor) -> torch.complex:
-    """Return the overlap (inner product) between two state vectors (kets).
+def ket_overlap(x: Tensor, y: Tensor) -> Tensor:
+    r"""Overlap (inner product) $\braket{\varphi,\psi}$ between two state vectors
+    $\ket\varphi$ and $\ket\psi$.
 
     Args:
-        x: Tensor of dimension `(..., d, 1)`.
-        y: Tensor of dimension `(..., d, 1)`.
+        x ( Tensor (..., d, 1) ): First state vector.
+        y ( Tensor (..., d, 1) ): Second state vector.
 
     Returns:
-        Complex-valued overlap (`torch.complex64` or `torch.complex128`).
+        ( Tensor (...) ): Complex-valued overlap.
     """
     return (ket_to_bra(x) @ y).squeeze(-1).sum(-1)
 
 
 def ket_fidelity(x: Tensor, y: Tensor) -> Tensor:
-    r"""Return the fidelity between two state vectors.
+    r"""Fidelity between two state vectors.
```

Comment:
```suggestion
    r"""Fidelity of two state vectors.
```

Reply:
QuTiP and Wikipédia use between, do you think it's uncorrect english?

Reply:
No it's because I looked at wikipedia and I think it uses "of" in https://en.wikipedia.org/wiki/Fidelity_of_quantum_states. It's "overlap between" but "fidelity of". I also think it sounds better. But whatever you prefer.